tools/logger/convert.c

```diff
@@ -349,16 +347,14 @@ static inline void print_table_header(void)
 
 	if (gettime_ret) {
 		log_err("clock_gettime() failed: %s\n",
-			strerror(gettime_ret));
+			strerror(errno));
 		exit(1);
 	}
 
 	if (global_config->time_precision >= 0) {
-		const unsigned int ts_width =
-			timestamp_width(global_config->time_precision);
-		snprintf(time_fmt, sizeof(time_fmt), "%%-%ds(us)%%%ds  ",
-			 ts_width, ts_width);
-		fprintf(out_fd, time_fmt, " TIMESTAMP", "DELTA");
+		const unsigned int ts_width = timestamp_width(global_config->time_precision);
+
+		fprintf(out_fd, "%*s(us)%*s  ", -ts_width, " TIMESTAMP", ts_width, "DELTA");
```

Comment:
No, this is not equivalent at all. We already discussed this:
- https://github.com/thesofproject/sof/pull/6738#discussion_r1042797977

This PR has clearly not been tested.

Reply:
In fact, it was done wrong in the linked PR. Run the code below and you will see that both approaches print the same result:

```
char time_fmt[255];
const unsigned int ts_width = 20;
    
snprintf(time_fmt, sizeof(time_fmt), "%%-%ds(us)%%%ds  ", ts_width, ts_width);
printf(time_fmt, " TIMESTAMP", "DELTA");
printf("\n");
printf("%*s(us)%*s  ", -ts_width, " TIMESTAMP", ts_width, "DELTA");
printf("\n");
```

Reply:
Sincere apologies, I had been "burned" by #6738 and this comment was rushed. I didn't know about `%*s` and it looks great.

Reply:
No problem, that's what I thought you did it automatically. We discover new things all our lives :)